File: frontend/management_app/src/interfaces/Product.ts
// Tipo para status do stock
export type StockStatus = "Normal" | "Baixo" | "Crítico" | "Esgotado";
// Tipo para categoria do produto
export type ProductCategory =
  | "Peças"
  | "Acessórios"
  | "Fluidos"
  | "Consumíveis"
  | "Ferramentas"
  | "Equipamentos"
  | "Outros";

// Interface para produto
export interface Product {
  id: string;
  partNumber: string;
  name: string;
  description: string;
  category: string;
  brand: string;
  quantity: number;
  reserveQuantity: number;
  costValue: number;
  saleValue: number;
  minimumStock: number;
  status: StockStatus;
}

File: backend/app/schemas/product.py
from pydantic import BaseModel
from typing import Optional


class ProductBase(BaseModel):
    partNumber: str;
    name: str;
    description: str;
    category: str;
    brand: str
    quantity: int
    reserveQuantity: Optional[int] = None
    costValue: float
    saleValue: float
    minimumStock: int
    
class ProductCreate(ProductBase):
    pass

class Product(ProductBase):
    id: int

    class Config:
        from_attributes = True